src/py42/clients/_settings_managers.py

```diff
@@ -0,0 +1,520 @@
+        new_val = self.get_converter(new_val)
+        if self.init_val is None:
+            init_val = get_val(instance.data, self.location)
+            self.init_val = self.get_converter(init_val)
+        if self.init_val == new_val:
+            if self.name in instance.changes:
+                instance.changes.pop(self.name)
+        else:
+            instance.changes[self.name] = show_change(self.init_val, new_val)
+
+
+class TSettingProperty(object):
+    """Descriptor class to help manage transforming t_setting packet values. Assumes t_setting
+    dict is stored in `._t_settings` attribute on managed instances.
+
+    Args:
+        name (str): name of attribute this class manages (changes will be registered with this name).
+        key (str): name of t_setting packet this class is managing.
+    """
+
+    def __init__(self, name, key):
+        self.name = name
+        self.key = key
+        self.init_val = None
+
+    def __get__(self, instance, owner):
+        packet = instance._t_settings[self.key]
+        return str_to_bool(packet["value"])
+
+    def __set__(self, instance, val):
+        val = bool_to_str(val)
+        packet = {"key": self.key, "value": val, "locked": False}
+        instance._packets[self.key] = packet
+        self._register_change(instance, val)
+
+    def _register_change(self, instance, val):
+        if self.init_val is None:
+            self.init_val = instance._t_settings[self.key]["value"]
+        if self.init_val == val:
+            if self.name in instance.changes:
+                instance.changes.pop(self.name)
+        else:
+            instance.changes[self.name] = show_change(self.init_val, val)
+
+
+class TrackedFileSelectionList(UserList):
+    """Helper class to track modifications to file selection lists."""
+
+    def __init__(self, manager, name, _list, changes_dict):
+        self.manager = manager
+        self.name = name
+        self.orig = list(_list)
+        self.data = _list
+        self._changes = changes_dict
+
+    def register_change(self):
+        self.manager._build_file_selection()
+        self.manager._build_regex_exclusions()
+        if set(self.orig) != set(self.data):
+            self._changes[self.name] = show_change(self.orig, self.data)
+        elif self.name in self._changes:
+            del self._changes[self.name]
+
+    def append(self, item):
+        self.data.append(item)
+        self.register_change()
+
+    def clear(self):
+        self.data.clear()
+        self.register_change()
+
+    def extend(self, other):
+        self.data.extend(other)
+        self.register_change()
+
+    def insert(self, i, item):
+        self.data.insert(i, item)
+        self.register_change()
+
+    def pop(self, index=-1):
+        value = self.data.pop(index)
+        self.register_change()
+        return value
+
+    def remove(self, value):
+        self.data.remove(value)
+        self.register_change()
+
+
+class OrgSettingsManager(UserDict):
```

Comment:
this plays into my other comment about the method param on DeviceClient etc...

Maybe these objects could actually be classified as mere `OrgSettings` rather than `OrgSettingsManager`.

Reply:
Yeah, I suppose the "manager" name made more sense when it had its own `.update()` method that actually made the http requests. I don't have a problem dropping it.